platform/platform_tests/duration_tests.cpp

```diff
@@ -0,0 +1,115 @@
+#include "testing/testing.hpp"
+
+#include "platform/duration.hpp"
+
+#include <chrono>
+
+namespace platform
+{
+using std::chrono::duration_cast, std::chrono::seconds, std::chrono::minutes, std::chrono::hours, std::chrono::days;
+
+struct TestData
+{
+  struct Duration {
+    days m_days;
+    hours m_hours;
+    minutes m_minutes;
+    std::string result;
+
+    Duration(long days, long hours, long minutes, std::string const & result)
+    : m_days(days), m_hours(hours), m_minutes(minutes), result(result)
+    {}
+
+    long Seconds() const
+    {
+      return (duration_cast<seconds>(m_days) +
+              duration_cast<seconds>(m_hours) +
+              duration_cast<seconds>(m_minutes)).count();
+    }
+  };
+
+  Locale m_locale;
+  std::vector<Duration> m_duration;
+
+  constexpr TestData(Locale locale, std::vector<Duration> duration)
+  : m_locale(locale), m_duration(duration)
+  {}
+};
+
+Locale GetLocale(std::string const & language)
+{
+  Locale locale;
+  locale.m_language = language;
+  return locale;
+}
+/*
+ Localized string cannot be retrieved from the app target bundle during the tests execution
+ and the platform::GetLocalizedString will return the same string as the input ("minute", "hour" etc).
+ This is why the expectation strings are not explicit.
+ */
+
+auto const m = Duration::GetUnitsString(Duration::Units::Minutes);
+auto const h = Duration::GetUnitsString(Duration::Units::Hours);
+auto const d = Duration::GetUnitsString(Duration::Units::Days);
+
+UNIT_TEST(Duration_AllUnits)
+{
+  auto const sg = kNonBreakingSpace;
+  TestData const testData[] = {
+    {GetLocale("en"),
+      {
+        {0, 0, 0,     "0" + m},
+        {0, 0, 1,     "1" + m},
+        {0, 0, 60,    "1" + h},
+        {0, 0, 123,   "2" + h + sg + "3" + m},
+        {0, 3, 0,     "3" + h},
+        {0, 24, 0,    "1" + d},
+        {4, 0, 0,     "4" + d},
+        {1, 2, 3,     "1" + d + sg + "2" + h + sg + "3" + m},
+        {1, 0, 15,    "1" + d + sg + "15" + m},
+        {0, 15, 1,    "15" + h + sg + "1" + m},
+        {1, 15, 0,    "1" + d + sg + "15" + h},
+        {15, 0, 10,   "15" + d + sg + "10" + m},
+        {15, 15, 15,  "15" + d + sg + "15" + h + sg + "15" + m}
+      }
+    },
+  };
+
+  for (auto const & data : testData)
+  {
+    for (auto const & dataDuration : data.m_duration)
+    {
+      auto const duration = Duration(dataDuration.Seconds());
+      auto durationStr = duration.GetLocalizedString({Duration::Units::Days, Duration::Units::Hours, Duration::Units::Minutes}, data.m_locale);
+      TEST_EQUAL(durationStr, dataDuration.result, ());
+
+      // Different order of units
+      durationStr = duration.GetLocalizedString({Duration::Units::Hours, Duration::Units::Days, Duration::Units::Minutes}, data.m_locale);
+      TEST_EQUAL(durationStr, dataDuration.result, ());
+    }
+  }
+}
+
+UNIT_TEST(Duration_Localization)
+{
+  auto const su = kNarrowNonBreakingSpace;
+  auto const sg = kNonBreakingSpace;
```

Comment:
Why such strange names? Using nbsp and nnbsp or full variables would be way clearer.